packages/w3/index.js

```diff
@@ -1,8 +1,71 @@
 import { globSource } from './utils/glob-source.js'
 import { writeFiles } from 'ipfs-car/unpack/fs'
 import { Web3Storage } from 'web3.storage'
+import enquirer from 'enquirer'
+import Conf from 'conf'
 import ora from 'ora'
 
+const API = 'https://api.web3.storage'
+const config = new Conf()
+
+/**
+ * Get a new API client configured either from opts or config
+ * @param {object} opts
+ * @param {string} [opts.api]
+ * @param {string} [opts.token]
+ */
+function getClient ({
+  api = config.get('api') || API,
+  token = config.get('token')
+}) {
+  if (!token) {
+    console.log('! run `w3 token` to set an API token to use')
+    process.exit(-1)
+  }
+  const endpoint = new URL(api)
+  if (api !== API) {
+    // note if we're using something other than prod.
+    console.log(`⁂ using ${endpoint.hostname}`)
+  }
+  return new Web3Storage({ token, endpoint })
+}
+
+/**
+ * Set the token and optionally the api to use
+ * @param {object} opts
+ * @param {string} [opts.api]
+ * @param {string} [opts.token]
+ */
+export async function token ({ token, api = API }) {
+  const url = new URL(api)
+  if (!token) {
+    const response = await enquirer.prompt({
+      type: 'input',
+      name: 'token',
+      message: `Paste your API token for ${url.hostname}`
+    })
+    token = response.token
+  }
+  config.set('token', token)
+  config.set('api', api)
+  console.log('⁂ API token saved')
+}
+
+/**
+ * Get data on Filecoin deals and IPFS pins that contain a given CID, as JSON.
```

Comment:
It would be nice to have a pretty print by default and an option to print as JSON.

Reply:
but JSON is pretty!